src/NzbDrone.Update/UpdateEngine/InstallUpdateService.cs

```diff
@@ -106,7 +109,31 @@ public void Start(string installationFolder, int processId)
             }
             finally
             {
-                _startNzbDrone.Start(appType, installationFolder);
+                if (OsInfo.IsWindows)
+                {
+                    _startNzbDrone.Start(appType, installationFolder);
+                }
+                else
+                {
+                    _terminateNzbDrone.Terminate(processId);
+
+                    _logger.Info("Waiting for external auto-restart.");
+                    for (int i = 0; i < 5; i++)
+                    {
+                        System.Threading.Thread.Sleep(i * 1000);
```

Comment:
On the other hand, it's a nice exponential back-off, checking quick in the beginning with a total of 15 sec.
Maybe that's what I intended when I wrote this yesterday, was kinda late.